tests/tests.cpp

```diff
@@ -42,33 +44,331 @@ Graph createGraph1() {
     return Graph(vect);
 }
 
-// Basic test cases
-TEST_CASE("initializeGraph1", "[weight=1][part=1]") {
-    // initialize and validate creating the graph with 5 test values
-    Graph graph = createGraph1();
+TEST_CASE("Test Find Path Lengths", "[weight=1][part=2]") {
+    Graph graph = prelimTestCase();
     REQUIRE(!(graph.getNodes()->empty()));
-}
-
-TEST_CASE("testGraph1", "[weight=1][part=2]") {
-    Graph graph = createGraph1();
     REQUIRE(graph.getNode("1") != nullptr);
     REQUIRE(graph.getNode("2") != nullptr);
     auto distances = graph.findPathLengths(graph.getNode("1"), 1);
     REQUIRE(!distances.empty());
-    /*std::cout << "test1" << std::endl;
-
-    for (unsigned int i = 0; i < distances.size(); i++) {
-        std::cout << i << " " << distances[i].second << std::endl;
-    }
-    
-    std::cout << "test2" << std::endl;*/
 
     REQUIRE(graph.findShortestPath(graph.getNode("1"), graph.getNode("2"), 1) == 1.0);
     REQUIRE(graph.findShortestPath(graph.getNode("1"), graph.getNode("3"), 1) == 3.0);
     REQUIRE(graph.findShortestPath(graph.getNode("1"), graph.getNode("4"), 1) == 6.0);
     REQUIRE(graph.findShortestPath(graph.getNode("1"), graph.getNode("5"), 1) == 5.0);
 }
 
+TEST_CASE("Dijkstra initialize", "[weight=1][part=2]") {
+    Graph* graph = new Graph("datasets/smallBasic.csv");
+    REQUIRE(!(graph->getNodes()->empty()));
+    REQUIRE(graph->getNode("1") != nullptr);
+    REQUIRE(graph->getNode("2") != nullptr);
+    REQUIRE(graph->getNode("3") != nullptr);
+
+    auto distances = graph->findPathLengths(graph->getNode("1"), 0);
+    REQUIRE(!distances.empty());
+}
+
+// Metric is 0 for fare (cost), 1 for time, and 2 for miles (length)
+TEST_CASE("Dijkstra small basic test", "[weight=1][part=2]") {
```

Comment:
ok, im trusting this works (@akuld2 ) i did see it pass, but just a question... what make you create your own graph instead of using one of the pre made ones?  was it because they weren't all ready yet. just a question

Reply:
oh wait im silly... ignore

Reply:
😎